sale_commission_partial_settlement/wizard/wizard_settle.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2023 Nextev
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, models
+
+
+class SaleCommissionMakeSettle(models.TransientModel):
+    _inherit = "sale.commission.make.settle"
+
+    def action_settle(self):
+        partial_res = self.action_settle_partial()
+        res = super().action_settle()
+        if partial_res and res:
+            partial_res["domain"][0][2] += res["domain"][0][2]
+            return partial_res
+        return res if res else partial_res
+
+    def action_settle_partial(self):
+        self.ensure_one()
+        settlement_obj = self.env["sale.commission.settlement"]
+        settlement_line_obj = self.env["sale.commission.settlement.line"]
+        settlement_ids = []
+
+        if self.agent_ids:
+            agents = self.agent_ids
+        else:
+            agents = self.env["res.partner"].search([("agent", "=", True)])
+        date_to = self.date_to
+        for agent in agents:
+            date_to_agent = self._get_period_start(agent, date_to)
+            partial_agent_lines, agent_lines_to_update = self.get_partial_agent_lines(
+                agent, date_to_agent, self.date_payment_to
+            )
+            for line_id in agent_lines_to_update:
+                self.env["account.invoice.line.agent"].browse(line_id).update(
+                    agent_lines_to_update[line_id]
+                )
+            for company in partial_agent_lines.mapped(
+                "invoice_line_agent_id.company_id"
+            ):
+                agent_lines_company = partial_agent_lines.filtered(
+                    lambda r: r.invoice_line_agent_id.object_id.company_id == company
+                )
+                pos = 0
+                sett_to = date(year=1900, month=1, day=1)
+                while pos < len(agent_lines_company):
+                    line = agent_lines_company[pos]
+                    pos += 1
```

Comment:
**thought**: this is not very pythonic.

I think there are more elegant ways of doing it, but at the very least

```py
sett_to = date.min
for line in agent_lines_company:
```